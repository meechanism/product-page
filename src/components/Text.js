import styled from 'styled-components';
import Colors from '../styles/colors';

export default styled.p`
  font-size: 1rem;
  line-height: 1.5rem;
  margin: 0 0 1rem;
  color: ${(props) => (props.muted ? Colors.gray : Colors.black)};
`;
