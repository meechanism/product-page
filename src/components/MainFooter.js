import React from 'react';
import styled from 'styled-components';

import Text from '../components/Text';

const FooterWrapper = styled.footer`
  padding: 2rem;
  text-align: center;
`;

const Footer = () => (
  <FooterWrapper>
    <Text>Â© 2020 Elmwest. All Rights Reserved.</Text>
    <Text>
      <a href="/">Terms and Conditions</a> | <a href="/">Privacy Policy</a> |{' '}
      <a href="/">Legal Statement</a>
    </Text>
  </FooterWrapper>
);

export default Footer;
