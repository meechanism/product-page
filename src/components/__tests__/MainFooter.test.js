import React from 'react';
import MainFooter from '../MainFooter';

import Text from '../Text';

describe('MainFooter component', () => {
  it('should render', () => {
    const output = shallow(<MainFooter />);
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should contain two texts', () => {
    const output = shallow(<MainFooter />);
    expect(output.find(Text).length).toBe(2);
  });
});
