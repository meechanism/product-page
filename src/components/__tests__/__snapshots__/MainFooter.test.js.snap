// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainFooter component should render 1`] = `
<styled.footer>
  <styled.p>
    Â© 2020 Elmwest. All Rights Reserved.
  </styled.p>
  <styled.p>
    <a
      href="/"
    >
      Terms and Conditions
    </a>
     | 
    <a
      href="/"
    >
      Privacy Policy
    </a>
     |
     
    <a
      href="/"
    >
      Legal Statement
    </a>
  </styled.p>
</styled.footer>
`;
